Formula/mongodb-csfle.rb

```diff
@@ -5,13 +5,11 @@ class MongodbCsfle < Formula
   license "MongoDB Customer Agreement"
 
   if Hardware::CPU.arm?
-    # TODO(MONGOSH-1193): Bump to production version after 6.0 is released
-    url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-arm64-enterprise-6.0.0-rc4.tgz"
-    sha256 "5e5bc88bcadf256c7dcea08a7aa55a552630de6e6919fcf3215b5e8171231c03"
+    url "https://downloads.mongodb.com/osx/mongo_csfle_v1-macos-arm64-enterprise-6.0.1.tgz"
```

Comment:
```suggestion
    url "https://downloads.mongodb.com/osx/mongo_crypt_shared_v1-macos-arm64-enterprise-6.0.1.tgz"
```

Reply:
Thanks for checking @addaleax! I updated in the latest rebase. You can close MONGOSH-1193 after this.